federation/cluster/federation-up.sh

```diff
@@ -81,6 +81,17 @@ function init() {
   kube::log::status "DNS_ZONE_NAME: \"${DNS_ZONE_NAME}\", DNS_PROVIDER: \"${DNS_PROVIDER}\""
   kube::log::status "Image: \"${kube_registry}/hyperkube-amd64:${kube_version}\""
 
+  # The very first thing that kubefed does when it comes up is run RBAC
+  # API discovery. We believe this sometimes fail on new clusters and as
+  # a result causes kubefed to assume that the RBAC API doesn't exist.
+  # Therefore, we are applying this workaround for now to ensure that the
+  # RBAC API is available before running kubefed.
+  while [[ ! "$(${KUBE_ROOT}/cluster/kubectl.sh api-versions)" =~ "rbac.authorization.k8s.io/" ]]; do
+    kube::log::status "Waiting for rbac.authorization.k8s.io API group to appear"
+    sleep 2
+  done
```

Comment:
Do you want to cap this to too many retries (kind of like how timeout does it below on kubefed), i.e.
timeout 1m sh -c "while [....]"
Just being defensive here since it's sometimes hard getting output out of federation-up.sh and timeout has a nice oddball exitval of 124